sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,395 @@
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(16, partitionId, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the partition processing task is stopping.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        ///
+        [Event(17, Level = EventLevel.Verbose, Message = "Stopping partition processing task for partition id '{0}' with reason '{1}'")]
+        public void StopPartitionProcessing(string partitionId, ProcessingStoppedReason reason)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(17, partitionId, reason);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while stopping a partition processing task.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        [Event(18, Level = EventLevel.Error, Message = "An exception occurred while stopping a partition processing task for partition id '{0}' with reason '{1}'. (Error Message: '{2}')")]
+        public void StopPartitionProcessingError(string partitionId, ProcessingStoppedReason reason, string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(18, partitionId, reason, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the partition processing task has been stopped.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        ///
+        [Event(19, Level = EventLevel.Verbose, Message = "Stopped partition processing task for partition id '{0}' with reason '{1}'")]
+        public void StoppedPartitionProcessing(string partitionId, ProcessingStoppedReason reason)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(19, partitionId, reason);
+            }
+        }
+
+        /// <summary>
+        ///   Signals the <see cref="EventProcessorClient" /> to stop processing events.
+        /// </summary>
+        ///
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        ///
+        [Event(20, Level = EventLevel.Verbose, Message = "Stop processing events. (Identifier '{0}')")]
```

Comment:
In general, we want `Informational` for those activities that would normally be helpful to establish context for those debugging.  Especially with the processor, a ton happens in the background places that are not visible to the developers using the library.   Our goal is to allow them to troubleshoot and more or less answer the questions "Is my code failing or is there a problem elsewhere"  and "Is there a problem with the client or is the service communication having issues."

Verbose would be for the things that may be situationally helpful/interesting in weird corner cases, but would otherwise be noise that interfered with the above questions.    If you're not sure which, I'd start with `Verbose` and we can always tweak.   It's easier to read less information than have to pick what you care about out of a wall of text.

Reply:
Thank you Jesse for the explanation. 

I've updated the `EventLevel` as per my understanding but like you said we can tweak if anything needs to be updated.